### DISCUSSION TITLE
Is nuxt-i18n-micro inspired from Vue i18n ?

### DISCUSSION THREAD

**Discussion by BaptisteCrouzet** [s00d/nuxt-i18n-micro#30] [General]
Hi, and first of all thank you for creating a performance alternative to nuxt-i18n.

When looking at this module, some questions came in my mind :
- as an alternative to nuxt-i18n, is it still even inspired by vue-i18n ?
- can we expect some features or api equivalences ? For example, randomly, modifiers https://vue-i18n.intlify.dev/guide/essentials/syntax.html
- is there some roadmap for that or any other new features you identified as missing ?

Thank you for your responses !
If something can be relevant to add to the doc, I would be happy to help !

**Comment by s00d** [comment]
Hi, and thank you for your interest in this project!

1. I wouldn’t exactly say it’s inspired by vue-i18n. My work started with optimizing and fixing certain issues in nuxt-i18n. I actually never looked into the original vue-i18n module itself. Unfortunately, the nuxt-i18n team considered my fixes non-critical, so I decided to create my own implementation from scratch. While some method names and parameters are similar, the underlying logic is entirely different and written by me. It’s a fresh approach, not a direct adaptation.

2. Regarding modifiers, I did implement them in one of the versions, but I ultimately realized they mostly duplicate what’s already possible with components like [`<i18n-link>`](https://s00d.github.io/nuxt-i18n-micro/components/i18n-link). With that component, you can easily replace anything you need. For example:
   
   ```vue
   <i18n-t keypath="feedback.text">
     <template #link>
       <nuxt-link :to="{ name: 'index' }">
         <i18n-t keypath="feedback.link" />
       </nuxt-link>
     </template>
   </i18n-t>
   ```
   
   This approach made me feel there wasn’t much value in releasing modifiers. I might add them in the future, but for now, I don’t see a strong need.

3. As for a roadmap, I don’t have any major plans for adding new features to the library itself. My focus is primarily on fixing issues when they arise. If you have ideas, I’m open to considering them, but for now, I feel the library covers most needs. That said, I am currently working on a translation service and plan to integrate it with the CLI utility. The service is mostly ready—just a few final tweaks—and I hope to release it soon.

And regarding the documentation, if you’re up for it, feel free to rewrite it as much as you’d like! Around 60% of it was generated with GPT since I’m not a fan of writing docs myself. It definitely has structural and descriptive issues that could be improved.

**Reply by BaptisteCrouzet** [reply]
Thank you for your responses ! As I will soon use more your module, I will help on what I find on the doc or what is missing for me ^^
